lib/connection/apm.js

```diff
@@ -190,17 +190,21 @@ class CommandSucceededEvent {
    * @param {Object} command the command
    * @param {Object} reply the reply for this command from the server
    * @param {Array} started a high resolution tuple timestamp of when the command was first sent, to calculate duration
+   * @param {Array} metadata an array of command metadata
    */
-  constructor(pool, command, reply, started) {
+  constructor(pool, command, reply, started, metadata) {
     const cmd = extractCommand(command);
     const commandName = extractCommandName(cmd);
+    metadata = (Array.isArray(metadata) && metadata) || [];
```

Comment:
@daprahamian how do you handle these internally? the alternative is to throw if a bad type is passed as metadata.

Reply:
updated